akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala

```diff
@@ -995,11 +1000,11 @@ private[cluster] class ClusterCoreDaemon(publisher: ActorRef) extends Actor with
    * Runs periodic leader actions, such as member status transitions, assigning partitions etc.
    */
   def leaderActions(): Unit = {
-    if (latestGossip.isLeader(selfUniqueAddress, selfUniqueAddress)) {
-      // only run the leader actions if we are the LEADER
+    if (latestGossip.isTeamLeader(myTeam, selfUniqueAddress, selfUniqueAddress)) {
+      // only run the leader actions if we are the LEADER (of the group)
       val firstNotice = 20
       val periodicNotice = 60
-      if (latestGossip.convergence(selfUniqueAddress, exitingConfirmed)) {
+      if (latestGossip.convergence(myTeam, selfUniqueAddress, exitingConfirmed)) {
```

Comment:
There is one log message around here that should be adjusted to only include members of own team. "Leader can currently not perform its duties..."